src/main/resources/templates/lista.html

```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+</head>
+<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css"
+      rel="stylesheet" integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1"
+      crossorigin="anonymous">
+<body>
+<div style="width: 250px; margin: 20px auto">
+    <h2>Lista produktów</h2>
+    <ul class="list-group">
+        <li class="list-group-item d-flex justify-content-between align-item-center"
+            th:each="product: ${products}">
+            <span th:text="|[${product.id}] ${product.name}|"></span>
```

Comment:
Jeżeli chcesz nadawać numer w tabeli takim produktom to możesz skorzystać z dodatkowej funkcjonalności thymeleafa przy iteracji - https://www.baeldung.com/thymeleaf-iteration#status-variable tutaj znajdziesz potrzebne Ci informacje. Wtedy możesz całkowicie zrezygnować z tego `id`